daemon/cmd/daemon_main.go

```diff
@@ -1662,6 +1662,7 @@ type daemonParams struct {
 	K8sResourceSynced      *k8sSynced.Resources
 	K8sAPIGroups           *k8sSynced.APIGroups
 	NodeManager            nodeManager.NodeManager
+	NodeHandler            datapath.NodeHandler
```

Comment:
Hmm, just realized that `NodeHandler` is implemented by multiple components, but here we're using it to explicitly mean the "linuxNodeHandler". This is super confusing. WDYT about adding a specific type for this? Maybe just `type LinuxNodeHandler datapath.NodeHandler` or `type DatapathNodeHandler datapath.NodeHandler`?

Could also make sense to move the `NodeHandler` interface into `pkg/node/manager`? Or I suppose alternatively we could have `datapath.NodeHandler` (implemented by linuxNodeHandler) and separately `manager.NodeHandler` that is relevant subset of `datapath.NodeHandler` that `NodeManager` invokes...

Reply:
Good point! But i'd like to refactor this in a separate PR. IMO `datapath.NodeHandler` is the right interface for depending on the datapath "nodehandler". It's up to the package `pkg/node/manager` to define its own version `manager.NodeHandler` that is used for subscribing to the `NodeManager`.